astropy/time/core.py

```diff
@@ -18,6 +22,11 @@
 import numpy as np
 import erfa
 
+if TYPE_CHECKING:
+    from typing_extensions import Self
+    import astropy.time
```

Comment:
This circular import seems strange but I assume is necessary?

Reply:
No it's not, Pycharm must've cleverly added it for me at some point when I erroneously typed "astropy.time.Time" somewhere. Good catch!